daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -121,7 +121,7 @@ private[lf] object Speedy {
     }
   }
 
-  final case class CachedContract(
```

Comment:
We now have a series of different types of internal caches, each with a specific intent. Perhaps it might be helpful to have a documentation string (e.g. at the top of `UpdateMachine` or similar) capturing information about what these caches are, how they should be used and when they should be updated?

Reply:
ok, good idea